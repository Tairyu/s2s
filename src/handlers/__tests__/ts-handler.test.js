// @flow
import handler from '../ts-handler'
import _plugin from './helpers/identifer-reverse-plugin'
import fromPlugin from './helpers/from-plugin'

const code = `
function greeter(person: string) {
  return "Hello, " + person;
}

let user = "Jane User";
greeter(user);
`

const plugin = {
  plugin: _plugin,
  test: /dummy/,
}

const opts = {
  eventPath: '/path/to/code.js',
  filename: 'dummy.js',
  plugin,
}

test('ts-handlerは変換後のコードを返す', () => {
  const result = handler(code, opts)
  expect(result).toMatchSnapshot()
})

test('pluginがArrayの場合', () => {
  const result = handler(code, {
    ...opts,
    plugin: { ...plugin, plugin: [_plugin, { x: 1 }] },
  })
  expect(result).toMatchSnapshot()
})

test('fromオプションを使う場合', () => {
  const result = handler(code, {
    ...opts,
    plugin: { ...plugin, plugin: fromPlugin },
  })
  expect(result).toMatchSnapshot()
})
