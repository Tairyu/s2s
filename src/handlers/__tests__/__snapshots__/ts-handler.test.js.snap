// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fromオプションを使う場合 1`] = `
"function greeterto(personto: string) {
  return \\"Hello, \\" + personto;
}

let userto = \\"Jane User\\";
greeterto(userto);"
`;

exports[`pluginがArrayの場合 1`] = `
"function reteerg(nosrep: string) {
  return \\"Hello, \\" + nosrep;
}

let resu = \\"Jane User\\";
reteerg(resu);"
`;

exports[`ts-handlerは変換後のコードを返す 1`] = `
"function reteerg(nosrep: string) {
  return \\"Hello, \\" + nosrep;
}

let resu = \\"Jane User\\";
reteerg(resu);"
`;
